explainaboard/loaders/file_loader.py

```diff
@@ -0,0 +1,221 @@
+            return field.parser(data)
+        if field.strip_before_parsing:
+            data = data.strip()
+        dtype = field.dtype
+        if dtype == FileLoaderDType.int:
+            return int(data)
+        if dtype == FileLoaderDType.float:
+            return float(data)
+        if dtype == FileLoaderDType.str:
+            return str(data)
+        if dtype == FileLoaderDType.other:
+            return data
+
+    def generate_id(self, parsed_data_point: dict, sample_idx: int):
+        """generates an id attribute for each data point in place"""
+        if self._use_idx_as_id:
+            parsed_data_point["id"] = str(sample_idx)
+        elif self._id_field_name:
+            if self._id_field_name not in parsed_data_point:
+                raise ValueError(
+                    f"The {sample_idx} data point in system outputs file does not have field {self._id_field_name}"
+                )
+            parsed_data_point["id"] = str(parsed_data_point[self._id_field_name])
+
+    @classmethod
+    def load_raw(self, data: str, source: Source) -> Iterable:
+        """Load data from source and return an iterable of data points. It does not use
+        fields information to parse the data points.
+
+        Args:
+            data (str): base64 encoded system output content or a path for the system output file
+                source: source of data
+        """
+        raise NotImplementedError(
+            "load_raw() is not implemented for the base FileLoader"
+        )
+
+    def load(self, data: str, source: Source) -> Iterable[dict]:
+        """Load data from source, parse data points with fields information and return an
+        iterable of data points.
+        """
+        raw_data = self.load_raw(data, source)
+        parsed_data_points: List[dict] = []
+        for idx, data_point in enumerate(raw_data):
+            parsed_data_point = {}
+
+            for field in self._fields:  # parse data point according to fields
+                parsed_data_point[field.target_name] = self.parse_data(
+                    data_point[field.src_name], field
+                )
+
+            self.generate_id(parsed_data_point, idx)
+            parsed_data_points.append(parsed_data_point)
+        return parsed_data_points
+
+
+@dataclass
+class FileLoaderField:
+    src_name: Union[int, str]  # int for tsv column indices and string for dict keys
+    target_name: str
+    dtype: FileLoaderDType
+    strip_before_parsing: bool = False
+    parser: Callable = None
+
+    def __post_init__(self):
+        for name in (self.src_name, self.target_name):
+            if not isinstance(name, str) and not isinstance(name, int):
+                raise ValueError("src_name and target_name must be str or int")
+        if self.dtype == FileLoaderDType.other and self.strip_before_parsing:
+            raise ValueError("dict type field does not support strip_before_parsing")
+
+
+class FileLoaderDType(str, Enum):
```

Comment:
Do we need this? Could we just use the standard Python typing mechansim?

Reply:
This is a great suggestion! I replaced the enum type with the python types (int, float, str, and None). None means other. I originally added an enum type because I thought we would need to support some more advanced types but I think these simple types cover everything we need now. If a user wants to work with some more complicated data schema, they can write their own parser pretty easily.